### PR TITLE
Pipeline Changes for Pushing Mirror and Build-Staging Repositories to Separate ACR

### PR DESCRIPTION
**Pull request by ellahathaway**
Related to https://github.com/dotnet/dotnet-docker-internal/issues/5024 and https://github.com/dotnet/dotnet-docker-internal/issues/5026

This PR updates the ymls to use the appropriate ACR server and serviceConnectionName for the given repo. The changes were made to the following ymls:

- common/templates/jobs/build-images.yml
- common/templates/jobs/publish.yml
- common/templates/steps/clean-acr-images.yml
- common/templates/steps/copy-base-images.yml
- common/templates/steps/test-images-linux-client.yml
- common/templates/steps/test-images-windows-client.yml
- pipelines/cleanup-acr-images-custom.yml
- pipelines/cleanup-acr-images.yml

[ImageBuilder pipeline run](https://dev.azure.com/dnceng/internal/_build/results?buildId=2459246&view=results)
[Windows tests pipeline run](https://dev.azure.com/dnceng/internal/_build/results?buildId=2462833&view=logs&j=ac422da1-c6cf-5423-8ede-22ab7060639a)
[Nightly pipeline run](https://dev.azure.com/dnceng/internal/_build/results?buildId=2465765&view=results)
[Cleanup pipeline run](https://dev.azure.com/dnceng/internal/_build/results?buildId=2465775&view=results)

### CODE REVIEW

File: eng/common/templates/steps/clean-acr-images.yml [hunk 1]
```diff
@@ -1,21 +1,23 @@
...
     parameters:
       displayName: Clean ACR Images - ${{ parameters.repo }}
-      serviceConnection: $(acr.serviceConnectionName)
+      serviceConnection: ${{ parameters.serviceConnection }}
       internalProjectName: ${{ parameters.internalProjectName }}
       args: >
         cleanAcrImages
         ${{ parameters.repo }}
         $(acr.subscription)
         $(acr.resourceGroup)
```
Comment (mthalman) [inline eng/common/templates/steps/clean-acr-images.yml:19]: This presumes the two ACRs will always be in the same subscription/resource group, which will likely not be the case in the future. I've gone ahead and created separate variables for this: `acr-staging.subscription` and `acr-staging.resourceGroup`. Please make use of those where appropriate.

**Comment by ellahathaway** [comment]
[New test run](https://dev.azure.com/dnceng/internal/_build/results?buildId=2462647&view=results) (internal Microsoft link)

**Comment by MichaelSimons** [comment]
Given @ellahathaway is offsite until Friday and then the short time-frame until patch Tuesday, it feels like we should defer merging until after Patch Tuesday.  Thoughts @mthalman and @lbussell?

**Comment by lbussell** [comment]
@michaelsimons I am OK with waiting since there is no immediate forcing function for these changes.

However, that would also mean we can't merge any other ImageBuilder changes all week, since the latest version requires these pipeline changes.

So, if we want to wait until after PT, we should either revert the source code changes, or adopt some version of these pipeline changes that don't push to the staging ACR. At which point we may as well just try out the whole set of changes.

I made the point in https://github.com/dotnet/docker-tools/pull/1295#issuecomment-2145579829 that we have a fair amount of time until Patch Tuesday. If @ellahathaway would like to be the one on point to fix potential issues from this change then like I said I'm OK with waiting.

**Comment by ellahathaway** [comment]
> If @ellahathaway would like to be the one on point to fix potential issues from this change then like I said I'm OK with waiting.

I'm happy to fix any potential issues that might arise, but like Michael mentioned, I won't be able to address anything until Friday of this week.

That said, I feel pretty confident in the changes in this PR (tested thoroughly with the specified pipelines in the PR description), but at the same time, I am not privy to all the repercussions that merging these changes might have.

Regardless, I'll take the lead in fixing any issues on Friday for whatever decision is made.

**Review by mthalman — APPROVED** [review]

**Comment by mthalman** [comment]
We might as well wait until after PT since these changes aren't urgent. I don't think there's a need to revert the other PR at this time. We can revert it if there's an Image Builder change we need to make in the mean time.

**Comment by mthalman** [comment]
Build is blocked. Need this fix: https://github.com/dotnet/docker-tools/pull/1330

**Comment by ellahathaway** [comment]
Build failed with:

`Unhandled exception: Azure.Identity.CredentialUnavailableException: DefaultAzureCredential failed to retrieve a token from the included credentials.`

This is the first time I'm seeing this error (didn't appear in my testing). Investigating.

**Pull request merged**